### ISSUE TITLE
Can we use Head-based sampling on iOS?

### ISSUE THREAD

**Issue by Mirconet** [DataDog/dd-sdk-ios#1713]
### Question

We want to use the [Head-based sampling](https://docs.datadoghq.com/tracing/guide/ingestion_sampling_use_cases/#solution-head-based-sampling), as suggested in your documentation, to keep the entire transaction traces. 
We managed to achieve the result in the Android app using the [dd-sdk-android](https://github.com/DataDog/dd-sdk-android). In the Android library, we found two different Sampler interfaces:

- one into [android-core](https://github.com/DataDog/dd-sdk-android/blob/6d24e01d7065137a7940db04d0e0b41e195ebe60/dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/sampling/Sampler.kt#L14)
- another into the [android-trace](https://github.com/DataDog/dd-sdk-android/blob/6d24e01d7065137a7940db04d0e0b41e195ebe60/features/dd-sdk-android-trace/src/main/java/com/datadog/trace/common/sampling/Sampler.java#L15)

The default implementation for the first sampler uses random sampling, while the implementation of the second uses deterministic sampling. Setting our custom sampling frequency only to the deterministic one and 100% to the other permits us to achieve the result on Android.

On the contrary on the DD iOS library, we saw that we could configure the sampling rate in two different points:
- on the `Trace.Configuration` init level
- on the `Trace.Configuration.URLSessionTracking` init level passed to the `Trace.Configuration` init

Both the settings seem to make the tracer use the same random [Sampler](https://github.com/DataDog/dd-sdk-ios/blob/develop/DatadogInternal/Sources/Utils/Sampler.swift) that doesn't permit us to link the spans between the chain of services and implement a Head-based sampling strategy to keep the entire transaction traces like Android.

How can we achieve this result on iOS?

**Comment by ncreated** [comment]
Hello @Mirconet 👋. Thank you for this report, I think this is indeed a problem. I added it to our backlog and we will discuss it in the team. We will keep you posted in this GH Issue.

I believe you're right that the 2 sampling rates available in `DatadogTrace` module are not in proper sync:
```swift
traceConfig.sampleRate = 50 // tracer sampling
traceConfig.urlSessionTracking = .init(
    firstPartyHostsTracing: .trace(hosts: ["foo.com"], sampleRate: 100) // distributed trace sampling
)
```
The **distributed trace sampling** should take precedence over **tracer sampling** in distributed tracing scenario. Meaning that if `foo.com` trace was sampled at 100%, the span sent from SDK must respect it. Instead it seems to use 50% configured in tracer sampler.

Unfortunately, I don't see any mitigation / workaround at this moment other than "no sampling" (100% in both). We will keep this thread with updates.

**Comment by ncreated** [comment]
Hey @Mirconet 👋. We just merged #1794 that adds support to head-based sampling. It should be available in next release.

**Comment by ncreated** [comment]
[2.11.0](https://github.com/DataDog/dd-sdk-ios/releases/tag/2.11.0) is out and it includes head-based sampling. Closing the issue.
